### ISSUE TITLE
bug: Multiple division assignment failing

### ISSUE THREAD

**Issue by neotheprogramist** [starkware-libs/cairo#4836]
# Bug Report

**Cairo version:**

v2.4.4

**Current behavior:**

When repeating the assignment `let value = 1 / 1;` 255 times or more in the code, the following error is encountered: 
    ```
    Error: #3585->#3586: Got 'Offset overflow' error while moving [3].
    ```

**Expected behavior:**

There should be no error or limit imposed on the number of times a division assignment can be executed in the code.

**Steps to reproduce:**

1. Clone the repository:
   ```
   git clone https://github.com/neotheprogramist/cairo-playground
   ```
2. Run the command:
   ```
   scarb cairo-run
   ```
   At this stage, the program should work correctly and return `1`.
3. Open `src/gen.cairo` and uncomment line `265`.
4. Run the command again:
   ```
   scarb cairo-run
   ```
   This time, the program fails and returns the error:
   ```
   Error: #3585->#3586: Got 'Offset overflow' error while moving [3].
   ```

**Related code:**

https://github.com/neotheprogramist/cairo-playground

**Other information:**

The issue does not occur when performing the operation in a loop, regardless the n

```cairo
let randoms1 = array![rand(), rand(), ..., rand()];
let randoms2 = array![rand(), rand(), ..., rand()];
let mut value = 0;
let mut i = 0;
loop {
    if i == n {
        break;
    }
    value = randoms1[i] / randoms2[i];
    i += 1;
};
```

**Comment by orizi** [comment]
This has nothing to do with the number of divisions happening in the code - this is all about the number of variables you define in the same scope - this division imposes a large AP-change, so you are bound by the number of such changes you perform.

Do you have an actual real world example where you ran into it?

**Comment by neotheprogramist** [comment]
@orizi 
Yes [here](https://github.com/HerodotusDev/cairo-verifier/blob/bf7de317211b6dbb91bc21de2816d054c6c0bc66/src/air/autogenerated.cairo#L1469-L3663)
I've needed to introduce a loop to do workaround.

**Comment by orizi** [comment]
In any case - even in generated code - i'd recommend against creating huge blocks of code especially if it creates thousands and thousands of variables that are all usable in the same scope.

**Comment by neotheprogramist** [comment]
@orizi 
Sure, we've already parsed that and created a loop - let's treat this issue only as a notification that such restriction exists though.

**Comment by shramee** [comment]
Hi Orizi,
I'm trying to write some tests for Fq6 (cubic over quadratic) such that I have Fq6 comprised of 3 Fq2 and finally the quadratic Fq2 of 2 Fq (represented by u256).

I have these tests,

```rs
#[test]
#[available_gas(5000000)]
fn mul() {
    // Having both of these causes this error:
    // #24531->#24532: Got 'Offset overflow' error while moving [62].
    let a = fq6(34, 645, 20, 55, 140, 105);
    let b = fq6(25, 45, 11, 43, 86, 101);
    let c = fq6(9, 600, 31, 12, 54, 4);

    // Having two of these cause the overflow error
    let ab = a * b;
    let bc = b * c;

    // This line line after the two above causes
    // Failed to cast from 63344.
    assert(ab * c == a * bc, 'incorrect mul');
}
 ```

I move ab and bc computation to separate scope without sharing any vars between them and it works with assert commented out. But of assert I need to share the previous result and I'm not able to avoid the error.

**Comment by orizi** [comment]
i'm assuming `fq6` is huge - can you try adding `core::internal::revoke_ap_tracking()` as the first statement in it?

**Comment by shramee** [comment]
Results in same error...

**Comment by shramee** [comment]
Aah that did it!!!! ❤️ ❤️ ❤️ ❤️ ❤️ ❤️ ❤️ 
Thanks so much!!

**Comment by orizi** [comment]
glad to help - i guess we need to track this and auto add this sort of thing.
as you seem to run into it more often lately.
